jax/lax.py

```diff
@@ -2233,12 +2232,63 @@ def _scatter_transpose_rule(t, operand, scatter_indices, updates,
                       slice_sizes=slice_sizes)
   return [operand_t, None, update_t]
 
+def _scatter_batching_rule(batched_args, batch_dims, update_jaxpr,
+                           update_consts, dimension_numbers, updates_shape):
+  operand, scatter_indices, updates = batched_args
+  operand_bdim, scatter_indices_bdim, updates_bdim = batch_dims
+  del update_jaxpr, update_consts, updates_shape  # Unused.
+
+  size = next(x.shape[ax] for x, ax in zip(batched_args, batch_dims)
+              if ax is not None)
+  operand = batching.bdim_at_front(operand, operand_bdim, broadcast_size=size,
```

Comment:
I don't follow this logic. Add a comment?

Reply:
Done!